.github/workflows/publish-to-npm.yaml

```diff
@@ -0,0 +1,25 @@
+name: Reusable Publish Package to npmjs workflow
+
+on:
+  workflow_call:
+    inputs:
+      workspace:
+        required: true
+        type: string
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+          registry-url: 'https://registry.npmjs.org'
+          scope: '@aws'
+      - run: npm ci
+      - run: npm run compile
+      - run: npm run pub --workspace ${{ inputs.workspace }}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
For time being using NPM_TOKEN stored in this repo secrets, not from AWS Secrets manager